pennylane/templates/state_preparations/basis_qutrit.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the BasisStatePreparation template.
+"""
+
+import pennylane as qml
+from pennylane.operation import Operation, AnyWires
+
+
+class QutritBasisStatePreparation(Operation):
+    r"""
+    Prepares a basis state on the given wires using a sequence of TShift gates.
+
+    .. warning::
+
+        ``basis_state`` influences the circuit architecture and is therefore incompatible with
+        gradient computations.
+
+    Args:
+        basis_state (array): Input array of shape ``(n,)``, where n is the number of wires
+            the state preparation acts on.
+        wires (Iterable): wires that the template acts on
+
+    **Example**
+
+    .. code-block:: python
+
+        dev = qml.device("default.qutrit", wires=4)
+
+        @qml.qnode(dev)
+        def circuit(basis_state, obs):
+            qml.QutritBasisStatePreparation(basis_state, wires=range(4))
+            return [qml.expval(qml.THermitian(obs, wires=i)) for i in range(4)]
+
+        basis_state = [0, 1, 1, 0]
+        obs = np.array([[1, 1, 0], [1, -1, 0], [0, 0, np.sqrt(2)]]) / np.sqrt(2)
```

Comment:
I have changed this so it is displayed as "code" rather than just a comment (to have it similar to the examples in other templates). I also changed `BasisStatePreparation` accordingly. I hope this is okay :)

Reply:
Nice!